app/controllers/CheckDetailsController.scala

```diff
@@ -79,44 +80,47 @@ class CheckDetailsController @Inject()(
       request.session.data.getOrElse("sessionId", EmptyString)
     )
 
-    val result: Try[Future[Result]] = Try {
-      val processData = for {
-        updatedAnswers <- Future.fromTry(request.userAnswers.set(OriginCacheable, origin.getOrElse("None")))
-        _ <- sessionRepository.set(updatedAnswers)
-        pdvData <- personalDetailsValidationService.getPDVData(pdvRequest)
-        idData  <- individualDetailsService.getIdData(pdvData)
-        sessionWithNINO = request.session + ("nino" -> getNinoFromPDVResponse(pdvData))
-      } yield (pdvData, idData, sessionWithNINO) match {
-        case (pdvData, Left(idData), sessionWithNINO)  => checkDetailsFailureJourney(pdvData, idData, mode, sessionWithNINO, origin)
-        case (pdvData, Right(idData), sessionWithNINO) => checkDetailsSuccessJourney(pdvData, idData, mode, sessionWithNINO, origin)
-        case _                                         => checkDetailsMatchingFailedWithUnknownIssue(mode)
-      }
-      processData.recover {
-        case ex: Exception =>
-          logger.error(s"An error occurred in process data: ${ex.getMessage}")
-          Redirect(routes.InvalidDataNINOHelpController.onPageLoad(mode = mode))
-      }
+    val processData:Future[Result] = for {
```

Comment:
Again linked to ITS-3543 this controller is getting too messy with multiple PDV checks and handling of PDV at multiple places in the code. 

I have an alternative suggestion that I want to run past the team in a mob?

Reply:
As I have created this branch on top of ITS-3543, we are seeing CheckDeatilsController changes but this is all about refactoring PDV validation status only.